test/e2e/loadbalancerservice_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"sigs.k8s.io/cloud-provider-huaweicloud/pkg/helper"
+	"sigs.k8s.io/cloud-provider-huaweicloud/test/e2e/framework"
+)
+
+var _ = ginkgo.Describe("loadbalancer service testing", func() {
+	var deployment *appsv1.Deployment
+	var service *corev1.Service
+	var secret *corev1.Secret
+
+	ginkgo.BeforeEach(func() {
+		deploymentName := deploymentNamePrefix + rand.String(RandomStrLength)
+		deployment = helper.NewDeployment(testNamespace, deploymentName)
+		framework.CreateDeployment(kubeClient, deployment)
+
+		secret = newSecret(testNamespace)
+		framework.CreateSecret(kubeClient, secret)
```

Comment:
The authentication of the HUAWEI CLOUD API will be used, and it must be created in the same namspace as the loadbalancer service.

Refer the code
https://github.com/kubernetes-sigs/cloud-provider-huaweicloud/blob/d686eefbfcd1f787fa813b4b6781ff11f422e124/pkg/cloudprovider/huaweicloud/alb.go#L445

I think ak and sk in [cloud-config](https://github.com/kubernetes-sigs/cloud-provider-huaweicloud/blob/master/examples/CloudControllerManagerConfigurationExample.json) can completely replace it, so I do some optimizations in #113 .

Reply:
The secret is used by `huawei-cloud-controller-manager`, not E2E test, it should be prepared before the E2E test.

In addition, it can't be present in `BeforeEach`, because it will be created before `each` test(ginkgo.It), we don't need to create it repeatedly.

Reply:
got it

Reply:
> The secret is used by `huawei-cloud-controller-manager`, not E2E test, it should be prepared before the E2E test.
> 
> In addition, it can't be present in `BeforeEach`, because it will be created before `each` test(ginkgo.It), we don't need to create it repeatedly.

This is a temporary solution, I will adjust it after I optimize the logic of obainting the secret.